src/pages/Florist/component/dialog.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import {
+  Button,
+  createStyles,
+  makeStyles,
+  Theme,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "@material-ui/core";
+
+import { Flower } from "../../../models/flower.model";
+import styles from "../Florist.module.css";
+
+interface AlertDialogProps {
+  selectFlower: Flower;
+  purchaseFunction: Function;
+  cancelFunction: Function;
```

Comment:
please define more specific types for functions, not type safe
could accept whatever function, 

`ie. : () => void` depends of the function interface you are passing as props